bootstrap.sh

```diff
@@ -47,7 +52,17 @@ env -C "$stage_4_dir" python3 -m src.glang.driver ./src/glang/parser/parser.c3 -
 
 # Stage 5; we introduce floating point literal support, compile the last parser
 # that supports this before literals become required by the standard library.
-env -C "$stage_5_dir" python3 -m src.glang.driver ./src/glang/parser/parser.c3 -o "$dest_dir/dist/parser"
+env -C "$stage_5_dir" python3 -m src.glang.driver ./src/glang/parser/parser.c3 -o "$stage_5_dir/dist/parser"
```

Comment:
```suggestion
env -C "$stage_5_dir" python3 -m src.glang.driver ./src/glang/parser/parser.c3 -o "$stage_6_dir/dist/parser"
```
Shouldn't you be building the _next_ stage's parser? And then you can change the Stage 6 command to use `env -C "$stage_6_dir"`

Reply:
The parser binary produced by stage 5 in incompatible with the python changes introduced in stage 6 (since the JSON output format has changed).

The highlighted stage fully bootstraps the stage 5 compiler (before it was in an intermediate state since it was running the stage 4 parser with the stage 5 python). This gives it just enough features to compile the new stage 6 parser with the magic file override.

Since this was the final bootstrap stage there was no need to run a fully bootstrapped stage 6 compiler

Reply:
The next stage looks awful since it was building the stage 6 parser+standard library with the stage 5 python. Note: the stage 6 python can't understand the old JSON format so it can't use the parser compiled as part of stage 5.

Reply:
Yeah, that makes sense. I don't like it (or how convoluted the bootstrap process is) but it makes sense.